### ISSUE TITLE
Validation and Evaluation Metric Issues in TensorFlow when Transfer Learning

### ISSUE THREAD

**Issue by Jon573** [tensorflow/tensorflow#35863]
I have come across some odd behaviours when training CNNs with Tensorflow 2.0 and would appreciate any help in solving them. I am doing transfer learning (just training the classification head) using the pre-trained networks available in 'tensorflow.keras.applications' and have noticed the following:

 1. For the first epoch, validation metrics are always zero, no matter what I do.  
 2. When training after the first epoch, the training metrics improve as you would expect, but the validation metrics essentially are random guesses, even when the EXACT same dataset is used as a training and a validation dataset. It is like it isn't using the model being trained to do its evaluation.

I have tried, VGG16, MobileNetV2, and ResNet50V2, and they all exhibit the same behaviours. 

The configurations I am able to reproduce this on are:
 - Ubuntu 18.04LTS, Nvidia RTX2080ti with driver version 430.50, CUDA10.0, TensorFlow-gpu==2.0.0 
 - MacBook Pro, TensorFlow==2.0.0 (cpu)

Both are running in Conda environments and I have installed TensorFlow with pip. I have put some sample code to show the essence of my workflow down below just in case I am doing anything obviously stupid. Any help would be very appreciated as I am at a loss as to how to fix it.

`
def parse_function(example_proto):
    image_feature_description = {
        'label': tf.io.FixedLenFeature([], tf.int64),
        'image_raw': tf.io.FixedLenFeature([], tf.string)
    }
    parsed_example = tf.io.parse_single_example(example_proto, image_feature_description)
    image = tf.io.decode_image(
                            parsed_example['image_raw'], 
                            channels = 3, 
                            dtype = tf.float32, 
                            expand_animations = False
                            )
    image = tf.image.per_image_standardization(image)
    label = tf.one_hot(parsed_example['label'], 24, dtype=tf.float32) 
    return (image, label)

def load_dataset(TFRecord_dir, record_name):
    record_files = tf.io.matching_files(os.path.join(TFRecord_dir, record_name + '.tfrecords-????'))
    shards = tf.data.TFRecordDataset(record_files)
    shards = shards.shuffle(tf.cast(tf.shape(record_files)[0], tf.int64))
    dataset = shards.map(map_func=parse_function)
    dataset = dataset.batch(batch_size=16, drop_remainder = True)
    dataset = dataset.prefetch(16)
    return dataset

base_model = tf.keras.applications.ResNet50V2(
                                            input_shape=(224,224,3),
                                            weights='imagenet',
                                            include_top = False
                                            )
base_model.trainable = False

model = tf.keras.Sequential([
        base_model,
        tf.keras.layers.GlobalAveragePooling2D(),
        tf.keras.layers.Dropout(0.5),
        tf.keras.layers.Dense(24, activation = 'softmax')
        ])

model.compile(
    optimizer=tf.keras.optimizers.Adam(),
    loss=tf.keras.losses.CategoricalCrossentropy(),
    metrics=[ 
            tf.keras.metrics.CategoricalAccuracy(),
            tf.keras.metrics.TopKCategoricalAccuracy(),
            tf.keras.metrics.Precision(),
            tf.keras.metrics.Recall()
            ])

train_dataset = load_dataset(train_dir, 'train')

model.fit(train_dataset,
                verbose = 1,
                epochs= 5,
                validation_data = train_dataset)
model.evaluate(train_dataset)
`

**Comment by oanush** [comment]
@Jon573 ,
Looks like the given code is incomplete,can you please share complete code?Thanks!

**Comment by Jon573** [comment]
Hi, thanks for your reply. All I have left out is:
`import tensorflow as tf` and `import os`, and `train_dir` is a folder on my local computer that contains TFRecords. I know the dataset is fine as I can use it with the Object Detection API and it works. I am using `train_dataset` as the training, validation and testing datasets as a debugging stage to rule out the model over fitting to the training dataset.

**Comment by oanush** [comment]
@Jon573 ,
Just provide us the complete code with link of data-set used so that we can reproduce the issue.Thanks!

**Comment by Jon573** [comment]
@oanush 
I am unable to provide the dataset for privacy reasons, but I no longer have the problem since I moved to using the latest docker image provided over the weekend. There must have been something wrong with the install on both machines but I don't know what. 

I also found, during debugging, that normalising the images using `image = (image/127.5) -1 ` as done in the transfer learning with trained CNN tutorial (https://www.tensorflow.org/tutorials/images/transfer_learning) would exhibit the same behaviour, even in the docker container, i.e. would training metrics would improve, but the validation metrics would remain random on the same dataset used to train.
